src/lib/request.ts

```diff
@@ -0,0 +1,48 @@
+import * as Bluebird from 'bluebird';
+import * as requestLib from 'request';
+import * as resumableRequestLib from 'resumable-request';
+
+import * as constants from './constants';
+import * as osRelease from './os-release';
+
+import supervisorVersion = require('./supervisor-version');
+
+const osVersion = osRelease.getOSVersionSync(constants.hostOSVersionPath);
+const osVariant = osRelease.getOSVariantSync(constants.hostOSVersionPath);
+
+let userAgent = `Supervisor/${supervisorVersion}`;
+if (osVersion != null) {
+	if (osVariant != null) {
+		userAgent += ` (Linux; ${osVersion}; ${osVariant})`;
+	} else {
+		userAgent += ` (Linux; ${osVersion})`;
+	}
+}
+
+// With these settings, the device must be unable to receive a single byte
+// from the network for a continuous period of 20 minutes before we give up.
+// (reqTimeout + retryInterval) * retryCount / 1000ms / 60sec ~> minutes
+const DEFAULT_REQUEST_TIMEOUT = 30000; // ms
+const DEFAULT_REQUEST_RETRY_INTERVAL = 10000; // ms
+const DEFAULT_REQUEST_RETRY_COUNT = 30;
+
+exports.requestOpts = {
```

Comment:
```suggestion
export const requestOpts = {
```

Reply:
Ah I missed this one, I've been using decaffeinate you see.